docker/docker_client_linux_test.go

```diff
@@ -0,0 +1,112 @@
+package docker_test
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"knative.dev/kn-plugin-func/docker"
+)
+
+func TestNewDockerClientWithAutomaticPodman(t *testing.T) {
```

Comment:
Please, if it's not to much trouble, can we please include at least a sentence comment on all tests?  it _really_ helps readers like me who are not familiar with the intention.  This is the "root" of this PR:  the statement of purpose.  Along with the test implementation itself this constitutes the expressed purpose of the whole feature, so having an English-natural-language description is not just helpful, it should really be the most basic expectation.